### ISSUE TITLE
xxx

### ISSUE THREAD

**Issue by madranet** [gilbertchen/duplicacy#238]
xxx

**Comment by madranet** [comment]
Is there a way to specify AWS Storage class when uploading? It defaults to Standard, where I want to use Standard-IA [infrequent access], which is cheaper and a better fit for a back up archive.

**Comment by gilbertchen** [comment]
Can you set up a transition rule to move chunk files from the standard storage to the infrequent access storage?

It should be easy to add a new storage url prefix, such as s3ia://, to support the infrequent access storage class, but I think if the transition rule works then this may not be necessary.

**Comment by madranet** [comment]
Yes. It's possible to add a rule to a bucket to transition from Standard to Standard-IA [which is what I've done]. The downside is that it's only possible to transition items to a different storage class, this way, after they've been in the bucket for 30 days. So that means that I'll be paying the more expensive rate for the first month on the entire original archive upload and on every subsequent incremental upload.

It would be helpful [and cheaper] if you could add an option to set storage class at initial upload.

**Comment by robbat2** [comment]
@gilbertchen you won't want all files in the same storage class blindly, and you do need to be able to set it per-file at upload time. Just adding another storage type/prefix cannot cover all the use cases.

- Standard-IA charges for at least 128KB; smaller files can be cheaper in Standard
- Standard-IA has extra costs for retrieval, so it's not ideal for that that might be fetched back often (configs, indices)
- Standard-IA charges for a min of 30 days; if you KNOW the file version will be replaced before that, it can be cheaper.

Glacier is similar, but 90 days, minimum charged size 1MB,  (they might have done away with it entirely, documentation is inconsistent), 32KB overhead (up from the normal 8KB overhead).

**Comment by gilbertchen** [comment]
This issue has been mentioned on **Duplicacy Forum**. There might be relevant details there:

https://forum.duplicacy.com/t/feature-request-support-s3-standard-infrequent-access-storage-class/4906/1
